samples/c/hello_classification/main.c

```diff
@@ -170,11 +170,8 @@ int main(int argc, char** argv) {
     c_mat_t img;
     image_read(input_image_path, &img);
     ov_element_type_e input_type = U8;
-    ov_shape_init(&input_shape, 4);
-    input_shape.dims[0] = 1;
-    input_shape.dims[1] = (size_t)img.mat_height;
-    input_shape.dims[2] = (size_t)img.mat_width;
-    input_shape.dims[3] = 3;
+    int64_t dims[4] = {1, (size_t)img.mat_height, (size_t)img.mat_width, 3};
+    ov_shape_init(&input_shape, 4, dims);
```

Comment:
After dimension and partial_shape are redefined to be a C style structure rather than an opaque handle, we can let user choose to define a variable or allocate a pointer varialable for partial_shape and dimension, so we move  structure itself allocation out of api and only initialize it in it.  To align, we put shape the similar behavior.

Reply:
ov_shape_init() should be more readable and meaningful, so as to avoid confusing user.

Reply:
Done - change all xxx_init() to xxx_create().